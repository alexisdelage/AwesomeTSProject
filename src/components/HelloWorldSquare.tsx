import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  square: {
    width: 200,
    aspectRatio: 1,
    backgroundColor: '#A02132',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginBottom: 50,
  },
  text: {
    color: '#70FF70',
    fontWeight: '900',
    fontSize: 50,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 'auto',
    marginBottom: 'auto',
  },
});

function HelloWorldSquare(): JSX.Element {
  return (
    <View style={styles.square}>
      <Text style={styles.text}>Hello!</Text>
    </View>
  );
}

export default HelloWorldSquare;
