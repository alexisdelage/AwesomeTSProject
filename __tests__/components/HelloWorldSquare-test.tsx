import React from 'react';
import renderer from 'react-test-renderer';
import HelloWorldSquare from '../../src/components/HelloWorldSquare';

it('renders correctly', () => {
  const tree = renderer.create(<HelloWorldSquare />).toJSON();
  expect(tree).toMatchSnapshot();
});
