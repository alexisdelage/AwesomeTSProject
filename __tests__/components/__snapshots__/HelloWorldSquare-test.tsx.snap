// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "aspectRatio": 1,
      "backgroundColor": "#A02132",
      "marginBottom": 50,
      "marginLeft": "auto",
      "marginRight": "auto",
      "width": 200,
    }
  }
>
  <Text
    style={
      {
        "color": "#70FF70",
        "fontSize": 50,
        "fontWeight": "900",
        "marginBottom": "auto",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "auto",
      }
    }
  >
    Hello
  </Text>
</View>
`;
